### ISSUE TITLE
orable object storage: multithread copy integration tests failing

### ISSUE THREAD

**Issue by ncw** [rclone/rclone#7457]
The new integration tests for oracle object storage are failing.

I recently made some new integration tests to test multipart copy (and I fixed the old tests so they actually tested the correct things). This has thrown up an error in the oracle object storage backend.

You can demonstrate this with the `fs/operations` tests

```
$ cd fs/operations
$ go test -v -timeout 1h0m0s -remote TestOracleObjectStorage: -verbose -test.run '^(TestMultithreadCopy|TestMultithreadCopyAbort)$'
```

There are two failing tests [TestMultithreadCopy](https://github.com/rclone/rclone/blob/b5857f0bf82238ea0b4a0a63825d3dd9c893668e/fs/operations/multithread_test.go#L145C17-L145C17) and [TestMultithreadCopyAbort](https://github.com/rclone/rclone/blob/b5857f0bf82238ea0b4a0a63825d3dd9c893668e/fs/operations/multithread_test.go#L263) which seem to fail in the same way, so I've only provided one output.

```
=== RUN   TestMultithreadCopy
    run.go:180: Remote "oos:bucket rclone-test-racaliv6fovamof8vexuguq1", Local "Local file system at /tmp/rclone3573138189", Modify Window "1ms"
    multithread_test.go:137: 
        	Error Trace:	/home/rclone/go/src/github.com/rclone/rclone/fs/operations/multithread_test.go:137
        	            				/home/rclone/go/src/github.com/rclone/rclone/fs/operations/multithread_test.go:148
        	Error:      	Received unexpected error:
        	            	create multipart upload request failed: Error returned by ObjectStorage Service. Http Status Code: 404. Error Code: BucketNotFound. Opc request id: lhr-1:a8PJ5X7Ty9K2Dw7tbvc5bbhij_rMhLLQawX0wKoTYljnupFU7GA31wJs3APCgNg7. Message: Either the bucket named 'rclone-test-racaliv6fovamof8vexuguq1' does not exist in the namespace 'lrqbljfginym' or you are not authorized to access it
        	            	Operation Name: CreateMultipartUpload
        	            	Timestamp: 2023-11-25 05:19:11 +0000 GMT
        	            	Client Version: Oracle-GoSDK/65.51.0
        	            	Request Endpoint: POST https://objectstorage.uk-london-1.oraclecloud.com/n/lrqbljfginym/b/rclone-test-racaliv6fovamof8vexuguq1/u
        	            	Troubleshooting Tips: See https://docs.oracle.com/iaas/Content/API/References/apierrors.htm#apierrors_404__404_bucketnotfound for more information about resolving this error.
        	            	Also see https://docs.oracle.com/iaas/api/#/en/objectstorage/20160918/MultipartUpload/CreateMultipartUpload for details on this operation's requirements.
        	            	To get more info on the failing request, you can set OCI_GO_SDK_DEBUG env var to info or higher level to log the request/response details.
        	            	If you are unable to resolve this ObjectStorage issue, please contact Oracle support and provide them this full error message.
        	Test:       	TestMultithreadCopy
--- FAIL: TestMultithreadCopy (0.22s)
```

Any ideas @msays2000 ?

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by msays2000** [comment]
@ncw will look into. I am also ramping up an internal dev to work on it.

**Comment by nikhilahuja** [comment]
@ncw Looks like the test doesn't create a test bucket during the test initialization or setup which leads to this error. When I run the same test with an existing bucket, it passes (see logs). What's the expectation of these Tests? Should it create a bucket before running? 
```
❯ go test -v -timeout 1h0m0s -remote remote:rclone-test -verbose -test.run '^(TestMultithreadCopy)$'
2023/11/27 17:25:51 DEBUG : Creating backend with remote "remote:rclone-test/rclone-test-fodined5pepizeb1hiremox1"
2023/11/27 17:25:51 DEBUG : Using config file from "/Users/nikhahuj/.config/rclone/rclone.conf"
2023/11/27 17:25:51 DEBUG : rclone-test-fodined5pepizeb1hiremox1: trying to read metadata rclone-test-fodined5pepizeb1hiremox1
2023/11/27 17:25:53 DEBUG : Creating backend with remote "/var/folders/gv/wcl33kqx5_q74z3plrtg1sz40000gn/T/rclone3482026975"
=== RUN   TestMultithreadCopy
    run.go:180: Remote "oos:bucket rclone-test, path rclone-test-fodined5pepizeb1hiremox1", Local "Local file system at /var/folders/gv/wcl33kqx5_q74z3plrtg1sz40000gn/T/rclone3482026975", Modify Window "1ms"
2023/11/27 17:25:54 DEBUG : chunksize-probe: created new upload id: e2340548-f874-0b3b-27a4-e4062cf1f729
2023/11/27 17:25:54 DEBUG : chunksize-probe: open chunk writer: started multipart upload: e2340548-f874-0b3b-27a4-e4062cf1f729
2023/11/27 17:25:54 DEBUG : chunksize-probe: Cancelling multipart upload
2023/11/27 17:25:54 DEBUG : chunksize-probe: canceled and aborted multipart upload: e2340548-f874-0b3b-27a4-e4062cf1f729
=== RUN   TestMultithreadCopy/upload=false,size=10485759,streams=2
2023/11/27 17:26:00 DEBUG : test-multithread-copy-false-10485759-2: trying to read metadata test-multithread-copy-false-10485759-2
2023/11/27 17:26:01 DEBUG : test-multithread-copy-false-10485759-2: trying to read metadata test-multithread-copy-false-10485759-2
2023/11/27 17:26:01 DEBUG : test-multithread-copy-false-10485759-2: trying to read metadata test-multithread-copy-false-10485759-2
2023/11/27 17:26:01 DEBUG : test-multithread-copy-false-10485759-2: trying to read metadata test-multithread-copy-false-10485759-2
2023/11/27 17:26:01 DEBUG : test-multithread-copy-false-10485759-2: multi-thread copy: disabling buffering because destination uses OpenWriterAt
2023/11/27 17:26:01 DEBUG : test-multithread-copy-false-10485759-2: multi-thread copy: write buffer set to 131072
2023/11/27 17:26:01 DEBUG : test-multithread-copy-false-10485759-2: multi-thread copy: chunk size 64Mi was bigger than source file size 10.000Mi
2023/11/27 17:26:01 DEBUG : test-multithread-copy-false-10485759-2: multi-thread copy: using backend concurrency of 4 instead of --multi-thread-streams 2
2023/11/27 17:26:01 DEBUG : test-multithread-copy-false-10485759-2: multi-thread copy: number of streams 4 was bigger than number of chunks 1
2023/11/27 17:26:01 DEBUG : test-multithread-copy-false-10485759-2: Starting multi-thread copy with 1 chunks of size 10.000Mi with 1 parallel streams
2023/11/27 17:26:01 DEBUG : test-multithread-copy-false-10485759-2: multi-thread copy: chunk 1/1 (0-10485759) size 10.000Mi starting
2023/11/27 17:26:02 DEBUG : test-multithread-copy-false-10485759-2: writing chunk 0
2023/11/27 17:26:05 DEBUG : test-multithread-copy-false-10485759-2: multi-thread copy: chunk 1/1 (0-10485759) size 10.000Mi finished
2023/11/27 17:26:05 DEBUG : test-multithread-copy-false-10485759-2: trying to read metadata test-multithread-copy-false-10485759-2
2023/11/27 17:26:05 DEBUG : test-multithread-copy-false-10485759-2: Finished multi-thread copy with 1 parts of size 10.000Mi
2023/11/27 17:26:05 DEBUG : test-multithread-copy-false-10485759-2: trying to read metadata test-multithread-copy-false-10485759-2
2023/11/27 17:26:06 DEBUG : test-multithread-copy-false-10485759-2: trying to read metadata test-multithread-copy-false-10485759-2
=== RUN   TestMultithreadCopy/upload=false,size=10485760,streams=2
2023/11/27 17:26:12 DEBUG : test-multithread-copy-false-10485760-2: trying to read metadata test-multithread-copy-false-10485760-2
2023/11/27 17:26:13 DEBUG : test-multithread-copy-false-10485760-2: trying to read metadata test-multithread-copy-false-10485760-2
2023/11/27 17:26:13 DEBUG : test-multithread-copy-false-10485760-2: trying to read metadata test-multithread-copy-false-10485760-2
2023/11/27 17:26:13 DEBUG : test-multithread-copy-false-10485760-2: trying to read metadata test-multithread-copy-false-10485760-2
2023/11/27 17:26:14 DEBUG : test-multithread-copy-false-10485760-2: multi-thread copy: disabling buffering because destination uses OpenWriterAt
2023/11/27 17:26:14 DEBUG : test-multithread-copy-false-10485760-2: multi-thread copy: write buffer set to 131072
2023/11/27 17:26:14 DEBUG : test-multithread-copy-false-10485760-2: multi-thread copy: chunk size 64Mi was bigger than source file size 10Mi
2023/11/27 17:26:14 DEBUG : test-multithread-copy-false-10485760-2: multi-thread copy: using backend concurrency of 4 instead of --multi-thread-streams 2
2023/11/27 17:26:14 DEBUG : test-multithread-copy-false-10485760-2: multi-thread copy: number of streams 4 was bigger than number of chunks 1
2023/11/27 17:26:14 DEBUG : test-multithread-copy-false-10485760-2: Starting multi-thread copy with 1 chunks of size 10Mi with 1 parallel streams
2023/11/27 17:26:14 DEBUG : test-multithread-copy-false-10485760-2: multi-thread copy: chunk 1/1 (0-10485760) size 10Mi starting
2023/11/27 17:26:14 DEBUG : test-multithread-copy-false-10485760-2: writing chunk 0
2023/11/27 17:26:15 DEBUG : test-multithread-copy-false-10485760-2: multi-thread copy: chunk 1/1 (0-10485760) size 10Mi finished
2023/11/27 17:26:15 DEBUG : test-multithread-copy-false-10485760-2: trying to read metadata test-multithread-copy-false-10485760-2
2023/11/27 17:26:15 DEBUG : test-multithread-copy-false-10485760-2: Finished multi-thread copy with 1 parts of size 10Mi
2023/11/27 17:26:15 DEBUG : test-multithread-copy-false-10485760-2: trying to read metadata test-multithread-copy-false-10485760-2
2023/11/27 17:26:16 DEBUG : test-multithread-copy-false-10485760-2: trying to read metadata test-multithread-copy-false-10485760-2
=== RUN   TestMultithreadCopy/upload=false,size=10485761,streams=2
2023/11/27 17:26:22 DEBUG : test-multithread-copy-false-10485761-2: trying to read metadata test-multithread-copy-false-10485761-2
2023/11/27 17:26:23 DEBUG : test-multithread-copy-false-10485761-2: trying to read metadata test-multithread-copy-false-10485761-2
2023/11/27 17:26:23 DEBUG : test-multithread-copy-false-10485761-2: trying to read metadata test-multithread-copy-false-10485761-2
2023/11/27 17:26:23 DEBUG : test-multithread-copy-false-10485761-2: trying to read metadata test-multithread-copy-false-10485761-2
2023/11/27 17:26:23 DEBUG : test-multithread-copy-false-10485761-2: multi-thread copy: disabling buffering because destination uses OpenWriterAt
2023/11/27 17:26:23 DEBUG : test-multithread-copy-false-10485761-2: multi-thread copy: write buffer set to 131072
2023/11/27 17:26:23 DEBUG : test-multithread-copy-false-10485761-2: multi-thread copy: chunk size 64Mi was bigger than source file size 10.000Mi
2023/11/27 17:26:23 DEBUG : test-multithread-copy-false-10485761-2: multi-thread copy: using backend concurrency of 4 instead of --multi-thread-streams 2
2023/11/27 17:26:23 DEBUG : test-multithread-copy-false-10485761-2: multi-thread copy: number of streams 4 was bigger than number of chunks 1
2023/11/27 17:26:23 DEBUG : test-multithread-copy-false-10485761-2: Starting multi-thread copy with 1 chunks of size 10.000Mi with 1 parallel streams
2023/11/27 17:26:23 DEBUG : test-multithread-copy-false-10485761-2: multi-thread copy: chunk 1/1 (0-10485761) size 10.000Mi starting
2023/11/27 17:26:24 DEBUG : test-multithread-copy-false-10485761-2: writing chunk 0
2023/11/27 17:26:25 DEBUG : test-multithread-copy-false-10485761-2: multi-thread copy: chunk 1/1 (0-10485761) size 10.000Mi finished
2023/11/27 17:26:25 DEBUG : test-multithread-copy-false-10485761-2: trying to read metadata test-multithread-copy-false-10485761-2
2023/11/27 17:26:25 DEBUG : test-multithread-copy-false-10485761-2: Finished multi-thread copy with 1 parts of size 10.000Mi
2023/11/27 17:26:25 DEBUG : test-multithread-copy-false-10485761-2: trying to read metadata test-multithread-copy-false-10485761-2
2023/11/27 17:26:25 DEBUG : test-multithread-copy-false-10485761-2: trying to read metadata test-multithread-copy-false-10485761-2
=== RUN   TestMultithreadCopy/upload=true,size=10485759,streams=2
2023/11/27 17:26:26 DEBUG : test-multithread-copy-true-10485759-2: multi-thread copy: disabling buffering because source is local disk
2023/11/27 17:26:27 DEBUG : test-multithread-copy-true-10485759-2: created new upload id: 3b042542-62da-6caa-8f52-5efc61faed37
2023/11/27 17:26:27 DEBUG : test-multithread-copy-true-10485759-2: open chunk writer: started multipart upload: 3b042542-62da-6caa-8f52-5efc61faed37
2023/11/27 17:26:27 DEBUG : test-multithread-copy-true-10485759-2: multi-thread copy: using backend concurrency of 10 instead of --multi-thread-streams 2
2023/11/27 17:26:27 DEBUG : test-multithread-copy-true-10485759-2: multi-thread copy: number of streams 10 was bigger than number of chunks 2
2023/11/27 17:26:27 DEBUG : test-multithread-copy-true-10485759-2: Starting multi-thread copy with 2 chunks of size 5Mi with 2 parallel streams
2023/11/27 17:26:27 DEBUG : test-multithread-copy-true-10485759-2: multi-thread copy: chunk 2/2 (5242880-10485759) size 5.000Mi starting
2023/11/27 17:26:27 DEBUG : test-multithread-copy-true-10485759-2: multi-thread copy: chunk 1/2 (0-5242880) size 5Mi starting
2023/11/27 17:26:27 DEBUG : test-multithread-copy-true-10485759-2: Seek from 5242879 to 0
2023/11/27 17:26:29 DEBUG : test-multithread-copy-true-10485759-2: Seek from 5242880 to 0
2023/11/27 17:26:30 DEBUG : test-multithread-copy-true-10485759-2: multi-thread copy: chunk 2/2 (5242880-10485759) size 5.000Mi finished
2023/11/27 17:26:32 DEBUG : test-multithread-copy-true-10485759-2: multi-thread copy: chunk 1/2 (0-5242880) size 5Mi finished
2023/11/27 17:26:33 DEBUG : test-multithread-copy-true-10485759-2: multipart upload 3b042542-62da-6caa-8f52-5efc61faed37 md5 matched: expecting LMcAd8vZUoM4i2joqnPkcQ==-2 and got LMcAd8vZUoM4i2joqnPkcQ==-2
2023/11/27 17:26:33 DEBUG : test-multithread-copy-true-10485759-2: trying to read metadata test-multithread-copy-true-10485759-2
2023/11/27 17:26:33 DEBUG : test-multithread-copy-true-10485759-2: Finished multi-thread copy with 2 parts of size 5Mi
2023/11/27 17:26:34 DEBUG : test-multithread-copy-true-10485759-2: Failed to read md5sum from metadata "LMcAd8vZUoM4i2joqnPkcQ==-2": illegal base64 data at input byte 24
2023/11/27 17:26:34 DEBUG : test-multithread-copy-true-10485759-2: trying to read metadata test-multithread-copy-true-10485759-2
2023/11/27 17:26:34 DEBUG : test-multithread-copy-true-10485759-2: trying to read metadata test-multithread-copy-true-10485759-2
=== RUN   TestMultithreadCopy/upload=true,size=10485760,streams=2
2023/11/27 17:26:35 DEBUG : test-multithread-copy-true-10485760-2: multi-thread copy: disabling buffering because source is local disk
2023/11/27 17:26:35 DEBUG : test-multithread-copy-true-10485760-2: created new upload id: 763119f2-d007-23b3-acca-eec1fd2cf789
2023/11/27 17:26:35 DEBUG : test-multithread-copy-true-10485760-2: open chunk writer: started multipart upload: 763119f2-d007-23b3-acca-eec1fd2cf789
2023/11/27 17:26:35 DEBUG : test-multithread-copy-true-10485760-2: multi-thread copy: using backend concurrency of 10 instead of --multi-thread-streams 2
2023/11/27 17:26:35 DEBUG : test-multithread-copy-true-10485760-2: multi-thread copy: number of streams 10 was bigger than number of chunks 2
2023/11/27 17:26:35 DEBUG : test-multithread-copy-true-10485760-2: Starting multi-thread copy with 2 chunks of size 5Mi with 2 parallel streams
2023/11/27 17:26:35 DEBUG : test-multithread-copy-true-10485760-2: multi-thread copy: chunk 2/2 (5242880-10485760) size 5Mi starting
2023/11/27 17:26:35 DEBUG : test-multithread-copy-true-10485760-2: multi-thread copy: chunk 1/2 (0-5242880) size 5Mi starting
2023/11/27 17:26:36 DEBUG : test-multithread-copy-true-10485760-2: Seek from 5242880 to 0
2023/11/27 17:26:36 DEBUG : test-multithread-copy-true-10485760-2: Seek from 5242880 to 0
2023/11/27 17:26:38 DEBUG : test-multithread-copy-true-10485760-2: multi-thread copy: chunk 1/2 (0-5242880) size 5Mi finished
2023/11/27 17:26:48 DEBUG : test-multithread-copy-true-10485760-2: multi-thread copy: chunk 2/2 (5242880-10485760) size 5Mi finished
2023/11/27 17:26:48 DEBUG : test-multithread-copy-true-10485760-2: multipart upload 763119f2-d007-23b3-acca-eec1fd2cf789 md5 matched: expecting A0OhxPIg/UwfZzlhAXOpaQ==-2 and got A0OhxPIg/UwfZzlhAXOpaQ==-2
2023/11/27 17:26:48 DEBUG : test-multithread-copy-true-10485760-2: trying to read metadata test-multithread-copy-true-10485760-2
2023/11/27 17:26:49 DEBUG : test-multithread-copy-true-10485760-2: Finished multi-thread copy with 2 parts of size 5Mi
2023/11/27 17:26:49 DEBUG : test-multithread-copy-true-10485760-2: Failed to read md5sum from metadata "A0OhxPIg/UwfZzlhAXOpaQ==-2": illegal base64 data at input byte 24
2023/11/27 17:26:49 DEBUG : test-multithread-copy-true-10485760-2: trying to read metadata test-multithread-copy-true-10485760-2
2023/11/27 17:26:50 DEBUG : test-multithread-copy-true-10485760-2: trying to read metadata test-multithread-copy-true-10485760-2
=== RUN   TestMultithreadCopy/upload=true,size=10485761,streams=2
2023/11/27 17:26:51 DEBUG : test-multithread-copy-true-10485761-2: multi-thread copy: disabling buffering because source is local disk
2023/11/27 17:26:51 DEBUG : test-multithread-copy-true-10485761-2: created new upload id: 8717fa2b-24be-2f71-6604-531fe369f4bc
2023/11/27 17:26:51 DEBUG : test-multithread-copy-true-10485761-2: open chunk writer: started multipart upload: 8717fa2b-24be-2f71-6604-531fe369f4bc
2023/11/27 17:26:51 DEBUG : test-multithread-copy-true-10485761-2: multi-thread copy: using backend concurrency of 10 instead of --multi-thread-streams 2
2023/11/27 17:26:51 DEBUG : test-multithread-copy-true-10485761-2: multi-thread copy: number of streams 10 was bigger than number of chunks 3
2023/11/27 17:26:51 DEBUG : test-multithread-copy-true-10485761-2: Starting multi-thread copy with 3 chunks of size 5Mi with 3 parallel streams
2023/11/27 17:26:51 DEBUG : test-multithread-copy-true-10485761-2: multi-thread copy: chunk 3/3 (10485760-10485761) size 1 starting
2023/11/27 17:26:51 DEBUG : test-multithread-copy-true-10485761-2: multi-thread copy: chunk 2/3 (5242880-10485760) size 5Mi starting
2023/11/27 17:26:51 DEBUG : test-multithread-copy-true-10485761-2: multi-thread copy: chunk 1/3 (0-5242880) size 5Mi starting
2023/11/27 17:26:51 DEBUG : test-multithread-copy-true-10485761-2: Seek from 1 to 0
2023/11/27 17:26:51 DEBUG : test-multithread-copy-true-10485761-2: Seek from 5242880 to 0
2023/11/27 17:26:52 DEBUG : test-multithread-copy-true-10485761-2: multi-thread copy: chunk 3/3 (10485760-10485761) size 1 finished
2023/11/27 17:26:52 DEBUG : test-multithread-copy-true-10485761-2: Seek from 5242880 to 0
2023/11/27 17:26:53 DEBUG : test-multithread-copy-true-10485761-2: multi-thread copy: chunk 1/3 (0-5242880) size 5Mi finished
2023/11/27 17:26:55 DEBUG : test-multithread-copy-true-10485761-2: multi-thread copy: chunk 2/3 (5242880-10485760) size 5Mi finished
2023/11/27 17:26:56 DEBUG : test-multithread-copy-true-10485761-2: multipart upload 8717fa2b-24be-2f71-6604-531fe369f4bc md5 matched: expecting miw+HpmMbt21ZU9G4F9wgQ==-3 and got miw+HpmMbt21ZU9G4F9wgQ==-3
2023/11/27 17:26:56 DEBUG : test-multithread-copy-true-10485761-2: trying to read metadata test-multithread-copy-true-10485761-2
2023/11/27 17:26:56 DEBUG : test-multithread-copy-true-10485761-2: Finished multi-thread copy with 3 parts of size 5Mi
2023/11/27 17:26:56 DEBUG : test-multithread-copy-true-10485761-2: Failed to read md5sum from metadata "miw+HpmMbt21ZU9G4F9wgQ==-3": illegal base64 data at input byte 24
2023/11/27 17:26:56 DEBUG : test-multithread-copy-true-10485761-2: trying to read metadata test-multithread-copy-true-10485761-2
2023/11/27 17:26:57 DEBUG : test-multithread-copy-true-10485761-2: trying to read metadata test-multithread-copy-true-10485761-2
--- PASS: TestMultithreadCopy (64.50s)
    --- PASS: TestMultithreadCopy/upload=false,size=10485759,streams=2 (12.03s)
    --- PASS: TestMultithreadCopy/upload=false,size=10485760,streams=2 (9.86s)
    --- PASS: TestMultithreadCopy/upload=false,size=10485761,streams=2 (9.69s)
    --- PASS: TestMultithreadCopy/upload=true,size=10485759,streams=2 (8.65s)
    --- PASS: TestMultithreadCopy/upload=true,size=10485760,streams=2 (15.72s)
    --- PASS: TestMultithreadCopy/upload=true,size=10485761,streams=2 (7.07s)
PASS
2023/11/27 17:26:58 DEBUG : oos:bucket rclone-test, path rclone-test-fodined5pepizeb1hiremox1: Purge dir ""
ok  	github.com/rclone/rclone/fs/operations	68.577s
```

**Comment by msays2000** [comment]
fixed it in https://github.com/rclone/rclone/pull/7462

```
go test -v -timeout 1h0m0s -remote TestOracleObjectStorage: -verbose -test.run '^(TestMultithreadCopy|TestMultithreadCopyAbort)$' -test.v
ok      github.com/rclone/rclone/fs/operations  65.346s
```

**Comment by msays2000** [comment]
@ncw please resolve this bug.

**Comment by ncw** [comment]
Thank you for fixing :-)

Putting `Fixes #7457` in the commit is my preferred way of resolving these but I didn't notice it was missing when  I reviewed!
